examples/digital_fingerprinting/production/Dockerfile

```diff
@@ -13,35 +13,83 @@
+    && bash /tmp/miniconda.sh -b -p /opt/conda \
+    && rm /tmp/miniconda.sh
+
+# Add conda to the PATH
+ENV PATH=/opt/conda/bin:$PATH
+
+# Install Mamba, a faster alternative to conda, within the base environment
+RUN conda install -y mamba -n base -c conda-forge
+
+# Create base environment
+RUN conda create -y --name morpheus-dfp python=3.10
+
+# Activate the environment (make it default for subsequent commands)
+RUN echo "source activate morpheus-dfp" >> ~/.bashrc
+
+# Set default shell to bash
 SHELL ["/bin/bash", "-c"]
 
-# Set the workdir to the DFP base folder
-WORKDIR /workspace/examples/digital_fingerprinting/
+# Default MORPHEUS_DFP_VERSION
+ARG MORPHEUS_DFP_VERSION=24.10.00a
+
+# Add conda channels required for the Morpheus dependencies
+RUN source activate morpheus-dfp \
+    && conda config --env --add channels conda-forge \
+    && conda config --env --add channels nvidia \
+    && conda config --env --add channels rapidsai \
+    && conda config --env --add channels pytorch
+
+# Install morpheus-dfp conda package
+RUN echo "MORPHEUS_DFP_VERSION is set to: ${MORPHEUS_DFP_VERSION}"
+ENV MORPHEUS_DFP_VERSION=${MORPHEUS_DFP_VERSION}
+RUN source activate morpheus-dfp \
+    && mamba install -y \
+     -c nvidia/label/dev morpheus-dfp=${MORPHEUS_DFP_VERSION}
+
+# Install the morpheus-dfp pip requirements
+RUN source activate morpheus-dfp \
+    && export MORPHEUS_DFP_PKG_DIR=$(dirname $(python -c "import morpheus_dfp; print(morpheus_dfp.__file__)")) \
+    && pip install -r ${MORPHEUS_DFP_PKG_DIR}/requirements_morpheus_dfp.txt
 
-# Install DFP dependencies
-RUN source activate morpheus \
-    && /opt/conda/bin/conda env update --solver=libmamba -n morpheus --file /workspace/conda/environments/examples_cuda-125_arch-x86_64.yaml
+RUN source activate morpheus-dfp \
+     && mamba install -y -c conda-forge tini=0.19
+
+ENTRYPOINT [ "/opt/conda/envs/morpheus-dfp/bin/tini", "--", "/workspace/examples/digital_fingerprinting/production/docker/entrypoint.sh" ]
+
+SHELL ["/bin/bash", "-c"]
 
 # Set the tracking URI for mlflow
 ENV MLFLOW_TRACKING_URI="http://mlflow:5000"
 
 # This will get used by pipelines for the --s3_cache option
 # ENV DFP_S3_CACHE="/work/examples/dfp_workflow/morpheus/.s3_cache"
 
-# If any changes have been made from the base image, recopy the sources
-COPY . /workspace/examples/digital_fingerprinting/
-
 # Set the workdir to be the production folder
 WORKDIR /workspace/examples/digital_fingerprinting/production
 
+# If any changes have been made from the base image, recopy the sources
+COPY . /workspace/examples/digital_fingerprinting/production/
+
+# Install additional dependencies needed to run the example.
+# Note: this call is very slow and is the longest part of the build proocess - fixme
+RUN source activate morpheus-dfp && /opt/conda/bin/conda env update --solver=libmamba -n morpheus-dfp --file ./conda/dfp_example_cuda-125_arch-x86_64.yaml
```

Comment:
Any reason why tini, morpheus-dfp, and the requirements aren't all rolled into the yaml? That would allow us to create the env with a single command.